File: Java/NavigationWithLogin/app/src/main/java/com/example/loginnavigation/MainActivity.java
package com.example.loginnavigation;

import android.net.Uri;
import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements RegisterFragment.OnFragmentInteractionListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity);
    }


    @Override
    public void onFragmentInteraction(Uri uri) {
        Log.i("MYTAG", "onFragmentInteraction uri " + uri);
    }
}

File: Kotlin/DataBindingDemo/app/src/main/java/com/example/databindingdemo/MainActivity.kt
package com.example.databindingdemo

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.databindingdemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.controlBtn.setOnClickListener {
            startStopProgress()
        }

        binding.openSeconAct.setOnClickListener {
            val intent = Intent(this, SecondActivity::class.java)
            startActivity(intent)
        }
    }

    private fun startStopProgress() {
        binding.apply {
            if (progressBar.visibility == View.GONE) {
                progressBar.visibility = View.VISIBLE
                controlBtn.text = "Start"
            } else {
                progressBar.visibility = View.GONE
                controlBtn.text = "Stop"
            }
        }
    }
}